src/backtrace.ts

```diff
@@ -0,0 +1,183 @@
+/**
+ * @packageDocumentation
+ * Collects backtraces for executables for diagnostics
+ */
+import { stringifyError } from "./errors.js";
+import * as actionsCore from "@actions/core";
+import * as exec from "@actions/exec";
+import { readFile, readdir } from "node:fs/promises";
+import { promisify } from "node:util";
+import { gzip } from "node:zlib";
+import os from "os";
+
+export async function collectBacktraces(
+  prefixes: string[],
+): Promise<Map<string, string>> {
+  if (os.platform() === "darwin") {
+    return await collectBacktracesMacOS(prefixes);
+  }
+  if (os.platform() === "linux") {
+    return await collectBacktracesSystemd(prefixes);
+  }
+
+  return new Map();
+}
+
+export async function collectBacktracesMacOS(
+  prefixes: string[],
+): Promise<Map<string, string>> {
+  const backtraces: Map<string, string> = new Map();
+
+  try {
+    const { stdout: logJson } = await exec.getExecOutput(
+      "log",
+      [
+        "show",
+        "--style",
+        "json",
+        "--last",
+        // Note we collect the last 1m only, because it should only take a few seconds to write the crash log.
+        // Therefor, any crashes before this 1m should be long done by now.
```

Comment:
s/Therefor/Therefore

Hyper-nitpicky, yes, but it is a public repo...

Reply:
Hell yeah nice